modules/openapi-generator/src/main/resources/python/model_utils.mustache

```diff
@@ -224,7 +225,8 @@ class OpenApiModel(object):
             self_inst.__init__(*args, **kwargs)
 
         new_inst = new_cls.__new__(new_cls, *args, **kwargs)
-        new_inst.__init__(*args, **kwargs)
+        # use _from_openapi_data to handle readOnly discriminator
+        new_inst = new_inst._from_openapi_data(*args, **kwargs)
```

Comment:
Once we get rid of the new invocation in the copy method can this code be reverted?
If not then I think there need to be two code paths here.
- For from_openapi_data == True, invoke _from_openapi_data
- For from_openapi_data == False invoke __new__ then __init__

Reply:
its the latter, its still needed (the openapi_data check). Agree, it would be more accurate to selectively call the correct init call. I will make that change.

Reply:
Since this is inside a __new__, trying to figure out if this is called from __init__ or from_openapi_data is tricky since __new__ only has a single arg (the class name).

Reply:
It seems the mechansim of validating the read-only vars at construction time (via init vs. from_openapi_data) leaves this code in a bit of a bind.

Reply:
Can we use kwargs['_spec_property_naming'] to decide which code path to use?
I think that
- _spec_property_naming = True is for the from_server use case / which is the from_openapi_data use case
- _spec_property_naming = False is the client side instantiation use case

Reply:
thats a good idea, I hadn't realized that _spec_property_naming had that property. However a click glance through the code, I don't see _spec_property_naming ever be se to True (its default is False in the _from_openapi_data). Is there another code path that is not obvious that sets this to True?

Reply:
Our code sets it to true here:
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/python/api_client.mustache#L340
when values are received from the server

Reply:
my concern is, this spec_property_naming is set via the deserialize call.
If I instantiate an object with _from_openapi_data(), and then copy that, I think that means now that I have to manually set this parameter (_spec_property_naming).

Reply:
Hmm that's true. Should all _from_openapi_data invocations hard code _spec_property_naming = True?

Reply:
possibly, though I'm not sure. My understanding is, _from_openapi_data is similiar to __init__, so vars are "pythonic". So that means it shouldn't be _spec_property_naming. I believe this tells me that we can't use_spec_property_naming for the purpose that I would like.

I don't quite appreciate the purpose of the _from vs. init. Couldn't the same thing be achieved and keep a single constructor initializer and just an option for checking read-only? I guess that is another discussion separate from this one.

Reply:
_from_openapi data is used to instantiate models with data received from the server. When making model instances with readonly parameters the fn signature needed to make a model instances must contain different parameters if the data came from the server vs from the client. You can read more about it here: https://github.com/OpenAPITools/openapi-generator/issues/9296#issuecomment-825768917
That conversation is separate from this one.
_from_openapi_data is used when instantiating models from the server so I'm pretty sure that we could depend on _spec_property_naming being true in it. We could throw an ApiValueError if _spec_property_naming is the arguments in _from_openapi_data.